### ISSUE TITLE
Error when shippingMethod is null

### ISSUE THREAD

**Issue by Kasoi-free** [tig-nl/postnl-magento2#50]
Magento 2.2.2
PostNL version 1.3.4

During extension settings we've spotted what might be an issue. 

_Sometimes_, when shipping method is not selected on checkout, PostNL extension throws an error below: https://screen.amasty.com/1526401504945.jpg

We've traced it to this file:
TIG_PostNL/js/Helper/State.js

This part:
```
quote.shippingMethod.subscribe(function (shippingMethod) {
        if (shippingMethod.carrier_code === 'tig_postnl') {
            return;
        }
        pickupAddress(null);
});
```

It monitors selected shipping method, but it does not have a workaround for situation when shippingMethod is not yet set, which, when this finally happens, generates an error shown in screenshot above.

**Comment by tig-jaspersmits** [comment]
Good morning,

Thank you for your submission. Is this the default Magento checkout? Because it looks like an one step checkout. We will most likely implement official support for 3rd party checkouts in the future.

Kind regards,

Jasper
TIG

**Comment by tig-dennisvanderhammen** [comment]
Hi,

As my colleague mentioned, we currently do not officially support 3rd party checkouts yet. Despite this we decided to pick up this issue. I've created a backlog item for this. You can follow the status of this issue on the following link:
https://jira.tig.nl/browse/POSTNLM2-543

**Comment by Kasoi-free** [comment]
@tig-jaspersmits we've got the issue to reproduce on Magento default checkout occasionally. That might be just a fluke though. On one step checkout solution we were getting more consistent results.

Possible workaround may look like this
```
  if (!!shippingMethod && shippingMethod.carrier_code === 'tig_postnl') {
   // some code
  }
```

**Comment by tig-dennisvanderhammen** [comment]
@Kasoi-free,

We added a fix for your issue in the latest version, v1.4.0.
Thank you for submitting your issue, and a possible workaround.
